crates/gitql-ast/src/function.rs

```diff
@@ -242,6 +250,25 @@ fn text_char(inputs: Vec<Value>) -> Value {
     return Value::Text("".to_string());
 }
 
+fn text_right(inputs: Vec<Value>) -> Value {
+    let text = inputs[0].as_text();
+    if text.is_empty() {
+        return Value::Text("".to_string());
+    }
+
+    let number_of_chars = inputs[1].as_number();
+    if number_of_chars > text.len() as i64 {
+        return Value::Text(text);
+    }
+    let substring = text
+        .chars()
+        .skip(text.len() - number_of_chars as usize)
+        .take(number_of_chars as usize)
+        .collect::<String>();
+
+    return Value::Text(substring);
```

Comment:
You can get sub string directly without collect, skip

```rs
Value::Text(text[number_of_chars - 1 .. text.len()].to_string())
```

Reply:
Yeahh, I'll check that out and correct it

Reply:
We have to take the characters from the right, not from the left, so, starting from `number_of_chars` will be wrong.